### DISCUSSION TITLE
Where to find version 5.4.17 (latest working on New Tapo cams)

### DISCUSSION THREAD

**Discussion by eavogels** [JurajNyiri/HomeAssistant-Tapo-Control#672] [Q&A] [answered]
Hi. Since TAPO CONTROL 5.4.17 was the last version that worked with cameras with new firmware (I have 3 HA instances that show my new Tapo cameras) I want to help my friend as well but how can I downgrade the HACS TAPO integration to 5.4.17? I do not find this older version.

**Comment by JurajNyiri** [comment, marked as answer]
Old integration versions do not fix new firmware issues.

**Reply by eavogels** [reply]
Well: I do not lie.   But anyhow: is 5.4.17 somewhere available? 

![image](https://github.com/user-attachments/assets/bedfe840-452d-4e95-88f3-6d36c2bd9f8e)

**Comment by eavogels** [comment]
I now bought and installed a new C200. I have newest firmware. I had to go back to Tapo Control 5.4.13 to get it to work with HA.
Perhaps you can compare code with 5.4.13 to find out where it broke?

![image](https://github.com/user-attachments/assets/93dd0b3d-72b0-4875-8701-b955b0d48f90)

**Reply by JurajNyiri** [reply]
That is impossible. This is caused by a cloud server Tapo change.

**Reply by JurajNyiri** [reply]
It must be something else than the cloud password change. Could you confirm which version breaks it for you? Strat going version by version and see when it breaks. Then please enable debug logs for the version that stops working and version that works and post them here.

**Reply by eavogels** [reply]
I have not activated the cloud solution on any of my Tapo cameras and I never did.  Some months ago I updated from the version I have to something newer, and I got that Cloud error message. I then did a complete restore and never tried to update again. I have 4 cameras now that do work in my Home Assistant and I don't want to break anything. The thing is that HACS want me to update from 5.4.17 to 5.6.3  and when I check the version of my integration, it shows 5.4.13

**Reply by JurajNyiri** [reply]
In that case 5.6.3 will most probably work. Stating old version fixes something you have not verified is actually caused by new integration version confuses users and wastes time of everyone.

You are using TPLink cloud if you have the camera, since you needed to create an account. Tapo care subscription is completely different thing and shall not be confused with cloud account.

If you have a fairly recent firmware, and TPLink chooses to push update for new credentials authorization to your account, any old, or new version of integration will break if the camera is connected to internet.
